### PR TITLE
Progress Reporting for long running operations

### PR DESCRIPTION
**Pull request by oakeyc**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by derekbekoe** [comment]
@oakeyc is this ready for re-review?

File: src/command_modules/azure-cli-interactive/azclishell/configuration.py [hunk 1]
```diff
@@ -63,6 +64,7 @@ def __init__(self):
         self.config.add_section('Layout')
         self.config.set('Help Files', 'command', 'help_dump.json')
         self.config.set('Help Files', 'history', 'history.txt')
+        self.config.set('Help Files', 'frequency', 'frequency.json')
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-interactive/azclishell/configuration.py]: This shouldn't be here right?

Comment (oakeyc) [inline src/command_modules/azure-cli-interactive/azclishell/configuration.py]: Yes, I removed it

File: src/azure-cli-core/azure/cli/core/commands/progress.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    def write(self, args):
+        """
+        writes the progress
+        :param args: args is a dictionary containing 'percent', 'message'
+        """
+        percent = args.get('percent', 0)
+        message = args.get('message', '')
+
+        if percent:
+            percent = percent
```
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/commands/progress.py]: This line does nothing right? Should remove if true.

Comment (oakeyc) [inline src/azure-cli-core/azure/cli/core/commands/progress.py]: You're right; I just removed it

File: src/azure-cli-core/azure/cli/core/commands/progress.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        self.active_progress = progress_view
+
+    def add(self, **kwargs):
+        """ adds a progress report """
+        self.reporter.add(**kwargs)
+        self.update()
+
+    def update(self):
+        """ updates the view with the progress """
+        # TODO APPROPRIATE PROGRESS SENT
```
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/commands/progress.py]: Remove todo?

File: src/command_modules/azure-cli-storage/tests/recordings/test_storage_blob_no_credentials_scenario.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+interactions:
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-storage/tests/recordings/test_storage_blob_no_credentials_scenario.yaml]: Was this added accidently?
Can you remove.

**Pull request merged**